columnphysics/icepack_therm_itd.F90

```diff
@@ -2016,7 +2016,9 @@ subroutine icepack_step_therm2(dt,           hin_max,       &
       ! Let rain drain through to the ocean.
       !-----------------------------------------------------------------
 
-      fresh  = fresh + frain * aice
+      if (.not.geos_heatflux) then  ! include rain drainage
+         fresh  = fresh + frain * aice
+      endif
```

Comment:
What does rain draining from the ice have to do with geos_heatflux?  This drainage is required if the level-ice or (new) sealvl pond schemes are used.  (The topo pond scheme is a bit more complicated.)

Reply:
This is more of legacy in GEOS coupling. We pass full rain flux directly to ocean, unlike other models doing the split (aice, 1-aice). geos_heatflux flag is just to limit this to GEOS. If this is not considered good practice, we may think about changing it in our coupler.

Reply:
With this rainwater treatment, then you should never turn on topo ponds (or in the future, any pond scheme that assumes rain water is collected).  I recommend aborting the run if this combination occurs.

Reply:
Thanks, @eclare108213. We sure don't want to lose the capability to use the topo pond scheme or others in future. I will consult with my team about how to move forward with it.

Reply:
Do we need another namelist variable to specifically control the fresh water coupling like "couple_rain" which is true by default and false for GEOS?  So rain does not fall on the sea ice but snow does?  It seems like some ice processes will be lost by doing this, right?  Would this be a good time to fix this in GEOS so rain is passed to CICE like snow?  Are there other sea ice models in GEOS that would be impacted by this coupling change?